src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpCertificateHelper.cs

```diff
@@ -74,5 +75,50 @@ internal static class CertificateHelper
                 }
             }
         }
+
+        public static X509Certificate2 GetEligibleClientCertificate()
+        {
+            // Get initial list of client certificates from the MY store.
+            X509Certificate2Collection candidateCerts;
+            using (var myStore = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            {
+                myStore.Open(OpenFlags.OpenExistingOnly | OpenFlags.ReadOnly);
+                candidateCerts = myStore.Certificates;
+            }
+            
+            return GetEligibleClientCertificate(candidateCerts);
+        }
+        
+        public static X509Certificate2 GetEligibleClientCertificate(X509Certificate2Collection candidateCerts)
+        {
+            if (candidateCerts.Count == 0)
+            {
+                return null;
+            }
+
+            // Reduce the set of certificates to match the proper 'Client Authentication' criteria.
+            candidateCerts = candidateCerts.Find(X509FindType.FindByKeyUsage, X509KeyUsageFlags.DigitalSignature, false);
+            candidateCerts = candidateCerts.Find(X509FindType.FindByApplicationPolicy, ClientAuthenticationOID, false);
+
+            // Build a new collection with certs that have a private key. Need to do this
+            // manually because there is no X509FindType to match this criteria.
+            var eligibleCerts = new X509Certificate2Collection();
+            foreach (var cert in candidateCerts)
+            {
+                if (cert.HasPrivateKey)
+                {
+                    eligibleCerts.Add(cert);
+                }
+            }
+            
+            if (eligibleCerts.Count > 0)
+            {
+                return eligibleCerts[0];
```

Comment:
Historically, we allow a collection of client certs for HttpWebRequest and HttpClientHandler. The collection is meant to be pared down to eligible certs and only the FIRST eligible cert is used and tried. There is no retry logic done.

The reason a collection was used in the first place was because the TrustedIssuers list was usually sent by a server to help further pare down the collection provided by the developer.  That collection a developer might provide to the API property would include a bunch of certs that could be used against server A but not server B. But it was easier for the developer to just set a collection once knowing that the "best" client cert would be tried (and only one would be tried).

We currently do NOT support retrieving the Trusted Issuers list from WinHTTP.  If that list is not sent by the server then all the certs would be considered eligible.  But still the architecture is to only try one cert.

I will open up a new issue to track adding the "Use Trusted Issuers List from WinHTTP".